packages/charts/src/chart_types/xy_chart/rendering/points.ts

```diff
@@ -112,7 +112,7 @@ export function renderPoints(
       const isInYDomain = yScale.isValueInDomain(valueAccessor(datum));
       indexedGeometryMap.set(pointGeometry, geometryType);
       // use the geometry only if the yDatum in contained in the current yScale domain
-      if (y !== null && yDefined(datum, valueAccessor) && isInYDomain && !isDatumFilled(datum)) {
+      if (isFiniteNumber(y) && yDefined(datum, valueAccessor) && isInYDomain && !isDatumFilled(datum)) {
```

Comment:
I don't think this check is necessary at all cuz canvas just ignores `NaN` values.

```suggestion
      if (yDefined(datum, valueAccessor) && isInYDomain && !isDatumFilled(datum)) {
```